libr/core/canal.c

```diff
@@ -2753,39 +2753,90 @@ R_API void r_core_recover_vars(RCore *core, RAnalFunction *fcn, bool argonly) {
 	return;
 }
 
-R_API RList* r_core_anal_graph_to(RCore *core, ut64 addr, int n) {
-	RAnalBlock *bb, *root = NULL, *dest = NULL;
-	RListIter *iter, *iter2;
-	RList *list2 = NULL, *list = NULL;
-	RAnalFunction *fcn;
+static int cmp_addr(const void *a, const void *b) {
+	const ut64 *_a, *_b;
+
+	_a = a;
+	_b = b;
+
+	return *_a - *_b;
+}
+
+static bool anal_path_exists(RCore *core, ut64 from, ut64 to, RList *bbs, int depth, RList *state) {
+	RAnalBlock *bb = r_anal_bb_from_offset (core->anal, from);
+	RListIter *iter = NULL;
+	RList *refs = NULL;
+	RAnalRef *refi;
+	RAnalFunction *fcn = NULL;
+	RAnalFunction *cur_fcn = NULL;
+	ut64 *frp = R_NEW0 (ut64);
+
+	if (depth < 1) {
+		eprintf ("going too deep\n");
+		return false;
+	}
+
+	if (!bb || !bbs) {
+		return false;
+	}
+
+	*frp = from;
+	r_list_append (state, (void *)frp);
+
+	ut64 tj = bb->jump;
+	ut64 fj = bb->addr + bb->size;
+
+	// try to find the target in the current function
+	if (r_anal_bb_is_in_offset (bb, to) ||
+		((!r_list_find (state, (void *)&(tj), cmp_addr) && anal_path_exists (core, bb->jump, to, bbs, depth - 1, state))) ||
+		((!r_list_find (state, (void *)&(fj), cmp_addr) && anal_path_exists (core, bb->addr + bb->size, to, bbs, depth - 1, state)))) {
+		r_list_prepend (bbs, bb);
+		return true;
+	}
 
+	// find our current function
```

Comment:
you can use `R_API RAnalFunction *r_anal_get_fcn_in(RAnal *anal, ut64 addr, int type);` to get the function `from` belongs to.

Reply:
I looked for something similar, but didn't look well enough apparently. thanks for the tip. will do that.

Reply:
done